how-it-works/7_smart_contracts/01_canister-lifecycle.subpage.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Canister smart contracts
+abstract: 
+shareImage: /img/how-it-works/canister-lifecycle.600.jpg
+slug: canister-lifecycle
+---
+
+# Canisters
+
+## What are canisters
+On the Internet Computer smart contracts come in the form of canisters.  These are computational units which bundle together code and state. Canisters expose endpoints which can be called both by other canisters and by parties external to the IC, such as browsers or mobile apps.  There are two types of endpoints. Updates are calls that can modify the state of the canisters and queries which cannot do that. A good mental model for these is that updates are used to write to the state of a canister and queries are used to read from that state.
+The code of a canister consists of a Wasm module. The state consists of the usual heap of the Wasm module, together with stable memory, a special type of memory which plays an important role in the life-cycle of a canister.
```

Comment:
I think we should write web assembly out at least for once and link to reference (either standards or Wikipedia or so). I'm also not sure how much "heap" is a "general audience" term.

Reply:
Fixed